stock_card_report/reports/stock_card_report_xlsx.py

```diff
@@ -0,0 +1,137 @@
+        for product in products:
+            for ws_params in self._get_ws_params(workbook, data, product):
+                ws_name = ws_params.get('ws_name')
+                ws_name = self._check_ws_name(ws_name)
+                ws = workbook.add_worksheet(ws_name)
+                generate_ws_method = getattr(
+                    self, ws_params['generate_ws_method'])
+                generate_ws_method(
+                    workbook, ws, ws_params, data, objects, product)
+
+    def _get_ws_params(self, wb, data, product):
+
+        stock_card_template = {
+            '1_date': {
+                'header': {
+                    'value': 'Date',
+                },
+                'data': {
+                    'value': self._render('date'),
+                    'type': 'datetime',
+                    'format': self.format_tcell_date_left,
+                },
+                'width': 25,
+            },
+            '2_reference': {
+                'header': {
+                    'value': 'Reference',
+                },
+                'data': {
+                    'value': self._render('reference'),
+                },
+                'width': 25,
+            },
+            '3_input': {
+                'header': {
+                    'value': 'Input',
+                },
+                'data': {
+                    'value': self._render('input'),
+                    'format': self.format_tcell_amount_right,
+                },
+                'width': 20,
+            },
+            '4_output': {
+                'header': {
+                    'value': 'Output',
+                },
+                'data': {
+                    'value': self._render('output'),
+                    'format': self.format_tcell_amount_right,
+                },
+                'width': 20,
+            },
+            '5_balance': {
+                'header': {
+                    'value': 'Balance',
+                },
+                'data': {
+                    'value': self._render('balance'),
+                    'format': self.format_tcell_amount_right,
+                },
+                'width': 20,
+            },
+        }
+
+        ws_params = {
+            'ws_name': product.name,
+            'generate_ws_method': '_stock_card_report',
+            'title': 'Stock Card Report - '+product.name,
+            'wanted_list': [k for k in sorted(stock_card_template.keys())],
+            'col_specs': stock_card_template,
+        }
+        return [ws_params]
+
+    def _stock_card_report(self, wb, ws, ws_params, data, objects, product):
+
+        ws.set_portrait()
+        ws.fit_to_pages(1, 0)
+        ws.set_header(self.xls_headers['standard'])
+        ws.set_footer(self.xls_footers['standard'])
+
+        self._set_column_width(ws, ws_params)
+
+        row_pos = 0
+        row_pos = self._write_ws_title(ws, row_pos, ws_params, True)
+
+        for o in objects:
+            ws.write_row(
```

Comment:
please write filter by used function _write_line
https://github.com/OCA/reporting-engine/pull/364

Reply:
@Saran440 What is the difference between write_row and _write_line ?